src/app/appointment/appointment.component.ts

```diff
@@ -50,6 +53,32 @@ export class AppointmentComponent implements OnInit, OnDestroy {
   }
 
   /**
+   * Sets the label for remaining time until next appointment
+   */
+  setCountdown(): void {
+    if (this.nextAppointments.length === 0) {
+      this.countdown = '';
+      return;
+    }
+
+    const timeDiff = this.getTimeDiff(this.nextAppointments[0]);
+
+    // remove current appointment from stack if it is in the past
+    if (timeDiff.asMinutes() < 0) {
+      this.nextAppointments.shift();
+      this.setCountdown();
+    }
+
+    this.countdown =
```

Comment:
I tried that, but it was quite inprecise. For example it showed "4 hours" when the appointment was in 4 hours and 30 minutes...